src/openfermion/utils/_trotter_qasm_conversion.py

```diff
@@ -0,0 +1,331 @@
+
+	Used in trotterize() function.
+
+	Params: 
+		trotterized_list: list that the function populates with trotterized
+				QubitOperators
+		op_a: first term in hamiltonian as a QubitOperator
+		op_b: the rest of the terms in the hamiltonian as a list of
+				QubitOperator term dictionary keys
+
+	Returns: None
+	'''
+	trotterized_list.append(7.0/24.0 * op_a)
+	if len(op_b) == 1:
+		trotterized_list.append(2.0 / 3.0 * QubitOperator(op_b[0],
+			hamiltonian_dict[op_b[0]]))
+	else:
+		btemp = op_b[1:]
+		for i in btemp:
+			hamiltonian_dict[i] *= 2.0 / 3.0
+		atemp = QubitOperator(op_b[0],hamiltonian_dict[op_b[0]])
+		atemp *= 2.0 / 3.0
+		third_order_trotter_helper(trotterized_list, atemp, btemp)
+	trotterized_list.append(3.0/4.0 * op_a)
+
+	if op_b.terms.size == 1:
+		trotterized_list.append(-2.0/3.0 * QubitOperator(op_b[0],
+			hamiltonian_dict[op_b[0]]))
+	else:
+		btemp = op_b[1:]
+		for i in btemp:
+			hamiltonian_dict[i] *= -2.0 / 3.0
+		atemp = QubitOperator(op_b[0],hamiltonian_dict[op_b[0]])
+		atemp *= -2.0 / 3.0
+		third_order_trotter_helper(trotterized_list,atemp,btemp)
+	trotterized_list.append(-1.0 / 24.0 * op_a)
+
+	if op_b.terms.size == 1:
+		trotterized_list.append(QubitOperator(op_b[0],
+			hamiltonian_dict[op_b[0]]))
+	else:
+		btemp = op_b[1:]
+		atemp = QubitOperator(op_b[0],hamiltonian_dict[op_b[0]])
+		third_order_trotter_helper(trotterized_list, atemp, btemp)
+
+
+
+
+def trotterize(
+	hamiltonian,trotter_steps = 1, 
+	trotter_order = 1, 
+	term_ordering = [], 
+	k_exp = 1.):
+	'''
+	This function trotterizes a hamiltonian to an order 1-3 and 
+	for a given number of trotter steps
+
+	Params:
+		hamiltonian: full hamiltonian as a QubitOperator.
+		trotter_steps: number of trotter steps as an integer.
+		order: order of trotterization as an integer from 1-3.
+		term_ordering (optional): list of tuples (QubitOperator terms
+				dictionary keys) that specifies order of terms when
+				trotterizing.
+		k_exp (optional): exponential factor to all terms when trotterizing.
+	
+	Returns:
+		tuple: (A list of single Pauli-term QubitOperators, the number of 
+		qubits required by the hamiltonian as an integer)
+	
+	Raises:
+		ValueError if order > 3 or order <= 0 or trotter_steps > 100,
+			TypeError for incorrect type 
+	'''
+	
+
+	if trotter_order > 3 or trotter_order <= 0:
+		raise ValueError("Invalid trotter order: " + str(order))
+	elif trotter_steps <= 0 or trotter_steps > 100: 
+		raise ValueError("Invalid number of trotter steps: " + str(trotter_steps))
+		
+	if not isinstance(hamiltonian, QubitOperator):
+		raise TypeError("Hamiltonian must be a QubitOperator.")
+	if not hamiltonian.terms:
+		raise TypeError("Hamiltonian must be a non-empty QubitOperator.")
+	if not isinstance(trotter_steps, int):
+		raise TypeError("trotter_steps must be an int")
+	if not isinstance(trotter_order, int):
+		raise TypeError("trotter_order must be an int")
+	if not isinstance(k_exp, float):
```

Comment:
I think some of the basic type checks aren't required. In particular, it's good to document the expected input type and enforce it for more specialized operators, but this particular check for k_exp being a float will fail for k_exp being an integer, which should be okay.  It will be naturally cast to float.  The same may be true for specialized numpy float types.